src/puppetlabs/puppetdb/queue.clj

```diff
@@ -30,15 +32,78 @@
     (catch Exception e
       (throw+ {:kind ::parse-error} e "Error parsing command"))))
 
+(defn sanitize-certname
+  "Replace any underscores and filename forbidden characters found in `certname`
+  with dashes."
+  [certname]
+  (let [forbidden-chars (conj constants/filename-forbidden-characters
+                              "_")]
+    (str/replace
+      certname
+      (re-pattern (str "("
+                       (->> forbidden-chars
```

Comment:
Since at the moment, we only use forbidden-chars here, I wondered if we might even consider just putting the right regex here directly, i.e. #"[/\_:]" or similar.